nat-lab/tests/telio.py

```diff
@@ -780,15 +878,23 @@ def get_features(self) -> TelioFeatures:
         assert self._telio_features
         return self._telio_features
 
-    async def stop_device(self, timeout: float = 5) -> None:
-        await asyncio.wait_for(self._write_command(["dev", "stop"]), timeout)
+    async def stop_device(self) -> None:
+        self.get_proxy().stop()
         self._interface_configured = False
-        started_tasks = self.get_runtime().get_started_tasks()
-        stopped_tasks = self.get_runtime().get_stopped_tasks()
-        diff = Counter(started_tasks) - Counter(stopped_tasks)
-        assert (
-            diff == Counter()
-        ), f"started tasks and stopped tasks differ! diff: {diff} | started tasks: {started_tasks} | stopped tasks: {stopped_tasks}"
+
+        # Check every .5s, up to maximum 10 seconds, that the started and stopped tasks are the same
```

Comment:
Could you clarify why do we need to wait now here?

Reply:
I can give a qualified guess. There was some overhead with how we were interacting with tcli that hid the time needed for telio to report stopped tasks, and now with the interaction with telio being synchronous, that delay is no longer there. We will likely never need to wait 10 seconds as specified here, but it's just a maximum. In most cases, it will just wait the first maybe 0.5s